### ISSUE TITLE
Ability to use Custom Models inside of Emotes, similiar to OriginRealms

### ISSUE THREAD

**Issue by Gebuildet** [PluginBugs/Issues-ItemsAdder#2319]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same suggestion was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if the feature is already implemented.
- [X] I already searched on the **Discord** server to know if anyone already has a solution for this or if the feature was already suggested.

### Discord tag (optional)

Gebuildet#0069

### Describe the solution you'd like

I would like to be able to add custom models to specific emotes in ItemsAdder, similiar to OriginRealms.

Example: 
![ezgif-3-02f6cc5028](https://user-images.githubusercontent.com/77403194/216551092-994c0c72-2f6d-4b0f-b152-2f3afb03a858.gif)

I would implement it in such a way that for each emote that contains a custom model, you create an extra emote file in which you then add the model.

### Is your feature request related to a problem?

Not being able to create Emotes, that are themed after specific stuff.

### Describe alternatives you've considered

There are no alternatives I can think of, since i don't know how you coded your plugin.

**Comment by LoneDev6** [comment]
Good idea, but would be very complex to implement with the current emotes structure.
I might implement this but it will be a bit tricky to use.

**Comment by LoneDev6** [comment]
# Temporary notes, update it not currently available publicly.

-----

# Notes

## Important

- You have to update the Blockbench extension to 1.2.0 to use this feature.
- DO NOT delete any builtin bone of the player.
- DO NOT resize any builtin bone of the player, it simply won't work.
- Edit builtin player bones only in the animation view (rotate, move)
- You cannot animate scaling of builtin player bones, but you can hide then using scale to `0,0,0`. Set to `1,1,1` to show again.

To use this feature you will have to update

## Extensions
- `.player_emotes` file format is now used to identify player animations which also add custom models to the emote.
- `.player_animations` are the player animations which don't add any custom model but will be loaded into the default player model.


In order to decide that you have to set this option in the Blockbench extension settings.
WARNING: you have to delete the folder if you switch from this option or it will contain also the file type and could create confusion.

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/c9ca9a77-bb79-4804-92e4-8708eac41517)

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/d6518a25-673b-42e4-8f79-fee8c07d7f02)

## Importing elements into the scene

You can import any `.bbmodel` into the scene and use it in your animation.
You can also import extruded textures to show items dynamically.

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/e8c72121-5eb7-4f44-88f9-31abd81380a4)


## Hiding unused elements in other animations of the game `.iaentitymodel` emotes file

1. select the animation
1. select the element you want to hide for a particular animation
2. move to the first frame of the animation, using the timeline
3. create a new scale key, make sure to set it to the first keyframe
4. set scale to 0

Now your model won't be shown for this particular animation.
You can show/hide it in the middle of the animation just by doing the same thing but setting scale to 1.

**Comment by LoneDev6** [comment]
# Tests

## Citizens

I tested using citizens and it works fine!

## ItemsAdder API

Works correctly! Example code:

```java
CustomPlayer customPlayer = CustomPlayer.spawn("LoneDev", location);
        LivingEntity entity = (LivingEntity) customPlayer.getEntity();
        entity.setAI(false);
        Scheduler.sync(() -> {
            customPlayer.playAnimation("no");
        }, 5 * 20);

        Scheduler.sync(() -> {
            customPlayer.playAnimation("amogus_kick");
        }, 10 * 20);
```

## ItemsAdder commands

Tested using `/iaentity` commands and they all work correctly.
